sycl/include/CL/sycl/handler.hpp

```diff
@@ -898,12 +898,38 @@ class __SYCL_EXPORT handler {
 
 #endif
 
+  std::shared_ptr<detail::kernel_bundle_impl>
+  getOrInsertHandlerKernelBundle(bool Insert);
+
+  void setHandlerKernelBundle(
+      const std::shared_ptr<detail::kernel_bundle_impl> &NewKernelBundleImpPtr);
+
 public:
   handler(const handler &) = delete;
   handler(handler &&) = delete;
   handler &operator=(const handler &) = delete;
   handler &operator=(handler &&) = delete;
 
+#if __cplusplus > 201402L
+  template <auto &SpecName>
+  void set_specialization_constant(
+      typename std::remove_reference_t<decltype(SpecName)>::type Value) {
```

Comment:
Note that the SYCL 2020 spec had a type.  The name `type` should instead be `value_type`.  See https://github.com/KhronosGroup/SYCL-Docs/pull/116

```suggestion
      typename std::remove_reference_t<decltype(SpecName)>::value_type Value) {
```